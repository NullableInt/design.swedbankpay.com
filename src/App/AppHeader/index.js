import React from "react";
import { Link, withRouter } from "react-router-dom";

import { GithubLogo, SlackLogo } from "./HeaderIcons";

import pkg from "../../../package.json";
import routes from "../routes/root.js";

const AppHeader = ({ location }) => {
    const { pathname } = location;

    return (
        <header className="topbar topbar-fixed designguide-header">
            <a className="topbar-logo" href="#"></a>
            <nav className="header-links">
                {routes.map((route, i) => {
                    const { title, path, exact } = route;
                    const active = exact ? (pathname === path) : (pathname.startsWith(path));

                    return <Link key={i} to={path} className={active ? "active" : ""}>{title}</Link>;
                })}
            </nav>
            <div className="topbar-info">
                <div className="topbar-info-contact">
                    <a href="https://github.com/PayEx/design.payex.com" target="b_blank" rel="noopener">
                        <GithubLogo />
                    </a>
                    <a href="https://payex.slack.com/messages/C0L3W8B2S/" target="b_blank" rel="noopener">
                        <SlackLogo />
                    </a>
                </div>
                <div className="topbar-info-version">
                    <a href="https://github.com/PayEx/design.payex.com/blob/master/CHANGELOG.md" target="b_blank" rel="noopener">v.{pkg.version}</a>
                </div>
            </div>
        </header>
    );
};

export default withRouter(AppHeader);
