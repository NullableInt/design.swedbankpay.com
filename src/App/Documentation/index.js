import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";

import SelectPanel from "./SelectPanel";
import DocHeading from "./utils/DocHeading";
import RenderRoutes from "./utils/RenderRoutes";
import routes from "../routes/docs";

const Documentation = () => (
    <div className="documentation">
        <div className="row">
            <div className="sidebar-heading col-xxl-2 col-md-3">
                <h1>DesignGuide</h1>
            </div>
            <div className="view-heading col-xxl-10 col-md-9">
                <DocHeading />
            </div>
        </div>
        <div className="row">
            <div className="col-xxl-2 col-md-3 p-0">
                <SelectPanel />
            </div>
            <main className="doc-view col-xxl-10 col-md-9">
                <Switch>
                    <Route exact path="/docs" render={() => <Redirect to="/docs/getting-started" />} />
                    {routes.map(route => <RenderRoutes key={`renderRoutes_${route.title}`} {...route} />)}
                    <Redirect from="/docs/*" to="/404" />
                </Switch>
            </main>
        </div>
    </div>
);

export default Documentation;
