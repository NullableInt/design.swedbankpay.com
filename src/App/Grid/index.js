import React, { Component } from "react";

class Examples extends Component {
    render () {
        return (
            <>
                <h1>GRID</h1>
                <div className="container">
                    <div className="row">
                        <div className="col-md-1 offset-md-2"></div>
                        <div className="col-md-2"></div>
                        <div className="col-md-3"></div>
                        <div className="col-md-4"></div>
                        <div className="col-md-5"></div>
                        <div className="col-md-6"></div>
                        <div className="col-md"></div>
                        <div className="col-md-auto"></div>
                    </div>
                </div>
            </>
        );
    }
}

export default Examples;
