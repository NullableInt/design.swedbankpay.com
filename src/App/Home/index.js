import React, { useEffect } from "react";
import { Link } from "react-router-dom";

import { setTitle } from "../utils";

const Home = () => {
    useEffect(() => {
        setTitle("Home");
    });

    return (
        <div className="container fp-container pt-3 pb-3">
            <div className="alert alert-warning">
                <header className="alert-header">
                    <i className="material-icons alert-icon">warning</i>
                    <h3>Site under development</h3>
                </header>
                <div className="alert-body">
                    <p>The Design Guide is under active development and thus subject to change.</p>
                </div>
            </div>
            <div className="row">
                <div className="col-12 col-md-4 pt-3 pt-md-0">
                    <div className="card card-plain">
                        <header className="card-header">
                            <h2>Placeholder</h2>
                        </header>
                        <div className="card-body">
                            <p>This paragraph contains some text.</p>
                        </div>
                    </div>
                </div>
                <div className="col-12 col-md-4 pt-3 pt-md-0">
                    <div className="card card-plain">
                        <header className="card-header">
                            <h2>Design Guide</h2>
                        </header>
                        <div className="card-body">
                            <div className="d-flex justify-content-center">
                                <i className="material-icons material-icons-large text-brand">computer</i>
                            </div>
                            <p>The Design Guide contains our component library for web development and general documentation for component use.</p>
                            <p>Check out the <Link to="/docs/getting-started/introduction">documentation</Link>.</p>
                        </div>
                    </div>
                </div>
                <div className="col-12 col-md-4 pt-3 pt-md-0">
                    <div className="card card-plain">
                        <header className="card-header">
                            <h2>Placeholder</h2>
                        </header>
                        <div className="card-body">
                            <p>This paragraph contains some text.</p>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Home;
