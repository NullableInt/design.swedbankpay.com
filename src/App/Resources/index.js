import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";

import SelectPanel from "../utils/SelectPanel";
import RenderRoutes from "../utils/RenderRoutes";
import routes from "../routes/resources";

const Resources = () => (
    <div className="documentation">
        <div className="row">
            <div className="col-xxl-2 col-md-3">
                <SelectPanel routes={routes} />
            </div>
            <main className="doc-view col-xxl-10 col-md-9">
                <Switch>
                    <Route exact path="/Resources" render={() => <Redirect to="/Resources/getting-started" />} />
                    {routes.map(route => <RenderRoutes key={`renderRoutes_${route.title}`} {...route} appFolder={"Resources"}/>)}
                    <Redirect from="/Resources/*" to="/404" />
                </Switch>
            </main>
        </div>
    </div>
);

export default Resources;
