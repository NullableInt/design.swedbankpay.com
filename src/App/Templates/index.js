import React, { Component } from "react";

import { setTitle } from "../utils";

class Templates extends Component {
    componentDidMount () {
        setTitle("Templates");
    }

    render () {
        return (
            <>
                <h1>Here there might be templates</h1>
            </>
        );
    }
}

export default Templates;
