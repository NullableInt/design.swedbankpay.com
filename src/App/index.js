import React, { Component } from "react";
import { BrowserRouter as Router, Switch, Route, withRouter } from "react-router-dom";

import routes from "./routes/root.js";
import AppHeader from "./AppHeader";
import ErrorPage404 from "./ErrorPage404";

const BASENAME = process.env.basename;

class ScrollToTop extends Component {
    componentDidUpdate (prevProps) {
        if (this.props.location !== prevProps.location) {
            window.scrollTo(0, 0);
        }
    }

    render () {
        return this.props.children;
    }
}

const ScrollToTopComponent = withRouter(ScrollToTop);

const App = () => (
    <div id="px-designguide">
        <Router basename={BASENAME}>
            <ScrollToTopComponent>
                <AppHeader />
                <Switch>
                    {routes.map((route, i) => {
                        const { path, component, exact } = route;
                        const RouteComponent = component.default;

                        return (
                            <Route key={`component_${i}`} exact={exact} path={path} render={() => {
                                if (process.env.google) {
                                    history.listen(location => {
                                        window.gtag("config", "UA-3440932-20", {
                                            /* eslint-disable camelcase */
                                            page_location: window.location.href,
                                            page_path: location.pathname
                                            /* eslint-enable camelcase */
                                        });
                                    });
                                }

                                return <RouteComponent />;
                            }} />
                        );
                    })}
                    <Route component={ErrorPage404} />
                </Switch>
            </ScrollToTopComponent>
        </Router>
    </div>
);

export default App;
