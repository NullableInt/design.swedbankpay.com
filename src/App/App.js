import React from "react";
import { BrowserRouter, withRouter, Switch, Route, Redirect } from "react-router-dom";

import AppHeader from "./AppHeader/Appheader";
import Home from "./Home/Home";
import Documentation from "./Documentation/Documentation";
import Examples from "./Examples/Examples";
import Templates from "./Templates/Templates";
import ErrorPage404 from "./ErrorPage404/ErrorPage404";

const AppHeaderWithRoutes = withRouter(props => <AppHeader {...props} />);

const App = () => {
    return (
        <BrowserRouter>
            <div className="doc-container" id="px-designguide">
                <AppHeaderWithRoutes />
                <Switch>
                    <Route exact path="/" render={() => (
                        <Redirect to="/home" />
                    )} />
                    <Route exact path="/home" component={Home} />
                    <Route path="/docs" component={Documentation} />
                    <Route path="/examples" component={Examples} />
                    <Route path="/templates" component={Templates} />
                    <Route path="/*" component={ErrorPage404} />
                </Switch>
            </div>
        </BrowserRouter>
    );
};

export default App;
