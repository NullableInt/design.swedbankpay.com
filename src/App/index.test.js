import React from "react";
import { shallow } from "enzyme";

import App from "./index";

const { topbar } = window.px;

describe("Main: App", () => {
    it("is defined", () => {
        expect(App).toBeDefined();
    });

    it("renders", () => {
        topbar.init = jest.fn();

        const wrapper = shallow(<App />);

        expect(wrapper).toMatchSnapshot();
        expect(topbar.init).toHaveBeenCalled();
    });
});
