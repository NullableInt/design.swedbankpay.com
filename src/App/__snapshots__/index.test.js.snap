// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main: App renders 1`] = `
<div
  id="px-designguide"
>
  <BrowserRouter
    basename="/TEST"
  >
    <withRouter(ScrollToTop)>
      <withRouter(AppHeader) />
      <Switch>
        <Route
          exact={true}
          key="component_0"
          path="/"
          render={[Function]}
        />
        <Route
          exact={false}
          key="component_1"
          path="/docs"
          render={[Function]}
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </withRouter(ScrollToTop)>
  </BrowserRouter>
</div>
`;
